packages/buckets/src/api/index.ts

```diff
@@ -631,8 +531,114 @@ export async function bucketsPushPathNode(
     stream.end()
   })
 }
+
+/**
+ * Pushes an iterable of files to a bucket.
+ * @param key Unique (IPNS compatible) identifier key for a bucket.
+ * @param input The input array of file/stream/objects.
+ * @param opts Options to control response stream.
+ * @internal
+ */
+export function bucketsPushPaths(
+  api: GrpcConnection,
+  key: string,
+  input: any,
+  opts?: Omit<PushOptions, 'progress'>,
+  ctx?: ContextInterface,
+): AsyncIterableIterator<PushPathsResult> {
+  return new Repeater<PushPathsResult>(async (push, stop) => {
```

Comment:
Simply returns an AsyncIterableIterator that only kicks off once you actually pull from it. This is nice, and makes it feel like a parallel push for an iterable of files. This function will take anything that can be converted to an async iterable of files... which includes iterables, async iterables, streams, etc.